site/app/models/RainbowCustomizationJSON.php

```diff
@@ -276,6 +294,23 @@ public function addMessage(string $message) {
         $this->messages[] = $message;
     }
 
+
+    /**
+     * Add plagiarism entry to existing array
+     *
+     * @param object $plagiarismEntry
+     */
+    public function addPlagiarismEntry(object $plagiarismEntry): void {
```

Comment:
Does this object have a more specific type?

Reply:
class PlagiarismEntry {
    public string $user;
    public string $gradeable;
    public int $penalty;

    public function __construct(string $user, string $gradeable, int $penalty) {
        $this->user = $user;
        $this->gradeable = $gradeable;
        $this->penalty = $penalty;
    }
}

Should I define new class for more specific type?

Reply:
public function __construct(string $user, string $gradeable, int $penalty) {
        $this->user = $user;
        $this->gradeable = $gradeable;
        $this->penalty = $penalty;
    }
Do you think if we should define new class to specify the type?

Reply:
I don't think there's a need for a whole data class here.  You can just specify the object shape with PHPDoc annotations and I'll be satisfied: https://phpstan.org/writing-php-code/phpdoc-types#object-shapes